py/fi_astrosims/client.py

```diff
@@ -55,36 +74,49 @@ def __init__(self, simulation, fields = None, sort = None, sample = 1, seed = No
+        if filters is not None:
+            for key in filters:
+                if 'transform' in simulation.fields[key]:
+                    filters[simulation.fields[key]['org-field']] = \
+                        tuple([x/simulation.fields[key]['scale'] for x in filters[key]])
+                    filters.pop(key)
         self.update(fields, sort, sample, seed, **filters)
 
     def update(self, fields = None, sort = None, sample = None, seed = None, **filters):
-        """Modify query parameters in place.
-
-        *fields* specifies the list of fields to return, defaulting to all
-        simulation fields.
-
-        *sort* specifies the ordering of result objects, which may be the name
-        of a single field, a field name prefixed with '-' for descending, or a
-        list of sort fields.
+        """
+        :param fields: (list) specifies the list of fields to return,
+        defaulting to all simulation fields.
+        :param sort: (type) specifies the ordering of result objects, which may
+        be the name of a single field, a field name prefixed with '-' for
+        descending, or a list of sort fields.
+        :param sample: (float) sets the fraction of results to return as a
+        random subset(0,1].
+        :param seed: (int) specifies the random seed for this selection, which
+        defaults to a different random set each time. If sample is set to a
+        value > 1, the value will be converted into a fraction within (0,1].
+        Note that each item is independently selected with the sample fraction,
+        so the result set may not be exactly this fraction of the original size.
+        :param filters: (dict) parameters specify restrictions on fields to
+        query by. Each parameter must be the name of a field in the catalog,
+        and have the value of None to not filter on this field, a scalar to
+        filter exact values, or tuple of lower- and upper-bounds to select
+        items in that range.
+        Sample is applied first, then the remaining filters.
+        """
 
-        *sample* sets the fraction of results to return as a random subset
-        (0,1].  *seed* specifies the random seed for this selection, which
-        defaults to a different random set each time.  Note that each item is
-        independently selected with the sample fraction, so the result set may
-        not be exactly this fraction of the original size.
+        # forces self.count() to return the max number of items in the catalog
+        self._count = None
+        self._aggs = {}
 
-        Additional *filters* parameters specify restrictions on fields to query
-        by.  Each parameter must be the name of a field in the catalog, and
-        have the value of None to not filter on this field, a scalar to filter
-        exact values, or tuple of lower- and upper-bounds to select items in
-        that range.
-        """
+        self.count()
```

Comment:
What's this count call doing? It'll cache it, but maybe should not. (count should be the count post-filtering)

Reply:
If I remember correctly, this count call forces the count to be the number of items in the entire dataset.